### PR TITLE
ttf

### PR DESCRIPTION
**Pull request by FredTheDino**
A quick and dirty implementation of font rendering.

TODOs that will come in another PR:
 - [ ] World position rendering of text
 - [ ] Layering - text is rendered on top of everything
 - [ ] Text rotation
 - [ ] Rewrite `push` to be the one-stop-shop for rendering

### CODE REVIEW

File: src/renderer.rs [hunk 1]
```diff
@@ -385,95 +389,21 @@ impl Renderer {
...
-                .assume();
-
-            if render.is_ok() {
-                context.window().gl_swap_window();
-                Ok(())
-            } else {
-                Err(())
-            }
-        };
-
```
Comment (FredTheDino) [inline src/renderer.rs:469]: I had a fit and moved this to the render function. Look at you past me!

File: res/LICENSE.txt [hunk 2]
```diff
@@ -0,0 +1,202 @@
+
+                                 Apache License
+                           Version 2.0, January 2004
+                        http://www.apache.org/licenses/
```
Comment (sornas) [inline res/LICENSE.txt:4]: Specify that this is the license for the font by e.g. renaming the license file to `LICENSE-noto-sans.txt`

File: src/renderer.rs [hunk 3]
```diff
@@ -385,95 +389,21 @@ impl Renderer {
...
             instances: vec![Vec::new()],
             tex,
             sprite_sheets: Vec::new(),
             particles: Vec::new(),
+            font: GlyphBrushBuilder::using_font(
+                // We forcefully include a default font,
+                // if you don't load any yourself.
+                // luminance_glyph requires ONE font.
+                FontArc::try_from_slice(include_bytes!("../res/noto-sans.ttf")).unwrap()
+                ).build(context),
```
Comment (sornas) [inline src/renderer.rs:403]:
```suggestion
            ).build(context),
```

File: src/main.rs [hunk 4]
```diff
@@ -128,6 +133,18 @@ fn main() {
...
+            .add_text(
+                Text::new("Hello Luminance")
+                .with_color([1.0, 1.0, 1.0, 1.0])
+                .with_scale(80.0))
+            .add_text(
+                Text::new("Glyph")
+                .with_color([1.0, 0.0, 0.0, 1.0])
+                .with_scale(80.0 + game.total_time().sin() * 20.0))
+        ;
+        game.renderer.push_text(section);
```
Comment (sornas) [inline src/main.rs:146]: How his position specified?

Comment (sornas) [reply src/main.rs:146]: I see now that you already noted this in the PR description, don't mind me.

**Review by sornas — APPROVED** [review]

**Pull request merged**